### PR TITLE
brew deps: support --full-name in --installed, --all, and --tree modes

### LINKED ISSUE #1692: deps command for all/installed formula does not support --full-name
**Bug/inconsistency report** (is improving consistency a feature request?)

`brew deps --installed` (or `--all`) ignores the `--full-name` argument. For example:

    $ brew deps --full-name netcdf
    homebrew/science/hdf5
    szip

Which is good and proper, but:

    $ brew deps --full-name | grep netcdf
    homebrew/science/netcdf: hdf5

The output should instead be:

    homebrew/science/netcdf: homebrew/science/hdf5

### PR DESCRIPTION
**Pull request by apjanke**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [X] Have you successfully run `brew tests` with your changes locally?

-----

Fixes #1692. See it for details.

Currently, the `--all`, `--installed`, and `--tree` modes ignore the `--full-name` option. This makes them respect it.

I haven't added new tests for this because I think that would require having a non-default tap tapped. The `tap_test.rb` file has code for setting up a tap, and that could be reused here with some refactoring, but that seems like overkill for this change.

### CODE REVIEW

File: Library/Homebrew/cmd/deps.rb [hunk 1]
```diff
@@ -66,16 +66,16 @@ def deps
     else
       all_deps = deps_for_formulae(ARGV.formulae, !ARGV.one?, &(mode.union? ? :| : :&))
       all_deps = all_deps.select(&:installed?) if mode.installed?
-      all_deps = if ARGV.include? "--full-name"
-        all_deps.map(&:to_formula).map(&:full_name)
-      else
-        all_deps.map(&:name)
-      end.uniq
+      all_deps = all_deps.map { |d| dep_display_name(d) }.uniq
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/deps.rb]: Think you can do `.map(&method(:dep_display_name)).uniq` here

File: Library/Homebrew/cmd/deps.rb [hunk 2]
```diff
@@ -127,7 +127,10 @@ def deps_for_formulae(formulae, recursive = false, &block)
   end
 
   def puts_deps(formulae)
-    formulae.each { |f| puts "#{f.full_name}: #{deps_for_formula(f).sort_by(&:name) * " "}" }
+    formulae.each do |f|
+      deps = deps_for_formula(f).sort_by(&:name).map { |d| dep_display_name(d) }
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/deps.rb]: Similarly here.

File: Library/Homebrew/cmd/deps.rb [hunk 3]
```diff
@@ -127,7 +127,10 @@ def deps_for_formulae(formulae, recursive = false, &block)
...
+      puts "#{f.full_name}: #{deps * " "}"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/deps.rb]: I think a `.join` is a bit more readable here but no strong preference.

**Comment by MikeMcQuaid** [comment]
> I haven't added new tests for this because I think that would require having a non-default tap tapped. The tap_test.rb file has code for setting up a tap, and that could be reused here with some refactoring, but that seems like overkill for this change.

Seems fine 👍

**Comment by apjanke** [comment]
Amended to address code comments and merged.

**Pull request merged**